extensions/wikia/CreateNewWiki/tasks/ConfigureWikiFactory.php

```diff
@@ -0,0 +1,262 @@
+			'wgSitename' => $siteName,
+			'wgLogo' => self::DEFAULT_WIKI_LOGO,
+			'wgUploadPath' => $imagesURL,
+			'wgUploadDirectory' => $imagesDir,
+			'wgDBname' => $dbName,
+			'wgLocalInterwiki' => $siteName,
+			'wgLanguageCode' => $language,
+			'wgServer' => rtrim( $url, "/" ),
+			'wgEnableSectionEdit' => true,
+			'wgEnableSwiftFileBackend' => true,
+			'wgOasisLoadCommonCSS' => true,
+			'wgEnablePortableInfoboxEuropaTheme' => true
+		];
+
+		// rt#60223: colon allowed in sitename, breaks project namespace
+		// Set wgMetaNamespace
+		if ( mb_strpos( $siteName, ':' ) !== false ) {
+			$wikiFactoryVariables['wgMetaNamespace'] = str_replace( [ ':', ' ' ], [ '', '_' ], $siteName );
+		}
+
+		if ( CreateDatabase::ACTIVE_CLUSTER ) {
+			wfGetLBFactory()->sectionsByDB[$dbName] = $wikiFactoryVariables['wgDBcluster'] = CreateDatabase::ACTIVE_CLUSTER;
+		}
+
+		return $wikiFactoryVariables;
+	}
+
+	public function getVariablesFromDB( $sharedDBW, $wikiFactoryVariables ) {
+		wfProfileIn( __METHOD__ );
+
+		$oRes = $sharedDBW->select(
+			"city_variables_pool",
+			[ "cv_id, cv_name" ],
+			[ "cv_name in ('" . implode( "', '", array_keys( $wikiFactoryVariables ) ) . "')" ],
+			__METHOD__
+		);
+
+		$wikiFactoryVarsFromDB = [ ];
+
+		while ( $oRow = $sharedDBW->fetchObject( $oRes ) ) {
+			$wikiFactoryVarsFromDB[$oRow->cv_name] = $oRow->cv_id;
+		}
+		$sharedDBW->freeResult( $oRes );
+
+		wfProfileOut( __METHOD__ );
+
+		return $wikiFactoryVarsFromDB;
+	}
+
+	private function setVariables( $sharedDBW, $cityId, $wikiFactoryVariablesFromDB, $wikiFactoryVariables ) {
+		wfProfileOut( __METHOD__ );
+		foreach ( $wikiFactoryVariables as $variable => $value ) {
+			/**
+			 * first, get id of variable
+			 */
+			$cv_id = 0;
+			if ( isset($wikiFactoryVariablesFromDB[$variable]) ) {
+				$cv_id = $wikiFactoryVariablesFromDB[$variable];
+			}
+
+			/**
+			 * then, insert value for wikia
+			 */
+			if ( !empty($cv_id) ) {
+				$sharedDBW->insert(
+					"city_variables",
+					array(
+						"cv_value" => serialize( $value ),
+						"cv_city_id" => $cityId,
+						"cv_variable_id" => $cv_id
+					),
+					__METHOD__
+				);
+			}
+		}
+
+		$sharedDBW->commit( __METHOD__ ); // commit shared DB changes
+		wfProfileOut( __METHOD__ );
+	}
+
+	/**
+	 * "calculates" the value for wgUploadDirectory
+	 *
+	 * @access private
+	 * @author Piotr Molski (Moli)
+	 *
+	 * @param $name string base name of the directory
+	 * @param $language string language in which wiki will be created
+	 *
+	 * @return string
+	 */
+	public function prepareDirValue( $name, $language ) {
+		wfProfileIn( __METHOD__ );
+
+		$this->debug( implode( ":", [ "CreateWiki", __METHOD__ . "Checking {$name} folder" ] ) );
+
+		$isExist = false;
+		$suffix = "";
+		$dir_base = self::sanitizeS3BucketName( $name );
+		$prefix = strtolower( substr( $dir_base, 0, 1 ) );
+		$dir_lang = (isset($language) && $language !== "en")
+			? "/" . strtolower( $language )
+			: "";
+
+		while ( $isExist == false ) {
```

Comment:
I had a similar case in CreateDatabase, maybe it would be a good idea to create some limit for while i.e. no more than a 100 checks, so that the code doesn't run forever if for some reason the dir checking returns true all the time.